sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -1346,4 +1345,118 @@ mod checked {
             protocol_config,
         )
     }
+
+    // Implementation of the `DataStore` trait for the Move VM.
+    // When used during execution it may have a list of new packages that have
+    // just been published in the current context. Thos are used for module/type
+    // resolution when executing module init.
+    // It may be created with an empty slice of packages either when no publish/upgrade
+    // are performed or when a type is requested not during execution.
+    pub(crate) struct SuiDataStore<'state, 'a> {
+        linkage_view: &'a LinkageView<'state>,
+        new_packages: &'a [MovePackage],
+    }
+
+    impl<'state, 'a> SuiDataStore<'state, 'a> {
+        pub(crate) fn new(
+            linkage_view: &'a LinkageView<'state>,
+            new_packages: &'a [MovePackage],
+        ) -> Self {
+            Self {
+                linkage_view,
+                new_packages,
+            }
+        }
+
+        fn get_module(&self, module_id: &ModuleId) -> Option<&Vec<u8>> {
+            for package in self.new_packages {
```

Comment:
What's our cap on new packages?

Reply:
it' a good question let me think about this a bit...

Reply:
made some changes, see if they make sense to you